packages/react-native-sdk/docusaurus/docs/reactnative/06-advanced/02-chat-with-video.mdx

```diff
@@ -3,6 +3,483 @@ title: Chat Integration
 ---
 
+It's common for calling apps to have chat, as well as the opposite - chat apps to have a calling functionality. Stream's Chat and Video SDKs are perfectly compatible between each other, and can easily be integrated into an app.
+
+Before starting with this guide on how to integrate chat and video together, we will initially focus on installation of the Chat SDK.
+
+## Installation
+
+To install Stream Chat SDK, you can go through our [Getting Started](https://getstream.io/chat/docs/sdk/reactnative/) guide and follow the installation steps and install all the relevant dependencies as well as optional dependencies according to your application's needs.
+
+:::note
+Do not forget through the [Additional Steps](https://getstream.io/chat/docs/sdk/reactnative/#additional-steps) guide while installing the dependencies.
+:::
+
+## Creating and Passing Chat Client
+
+To create the Chat Client we will mainly follow [this tutorial](https://getstream.io/chat/react-native-chat/tutorial/#add-stream-chat-to-the-application).
+
+### Wrapping the chat client
+
+To wrap our Chat initiation and client logic we will create a `ChatWrapper` component that can be wrapped at the root of the application, so that the client will be available to all the child components.
+
+```tsx title="ChatWrapper.tsx"
+import React, { PropsWithChildren, useCallback, useMemo } from 'react';
+import { Chat, OverlayProvider, Streami18n } from 'stream-chat-react-native';
+import { useChatClient } from '../hooks/useChatClient';
+import { AuthenticationProgress } from './AuthenticatingProgress';
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import { StyleSheet } from 'react-native';
+
+const streami18n = new Streami18n({
+  language: 'en',
+});
+
+export const ChatWrapper = ({ children }: PropsWithChildren<{}>) => {
+  const user = {
+    id: 'your-user-id',
+    name: 'your-user-name',
+  };
+  const token = 'your-user-token';
+
+  const chatClient = useChatClient({
+    apiKey: STREAM_API_KEY,
+    userData: user,
+    tokenProvider: token,
+  });
+
+  if (!chatClient) {
+    // Show a loader until the Chat client loads
+    return <AuthenticationProgress />;
+  }
+
+  return (
+    <GestureHandlerRootView style={styles.container}>
+      <OverlayProvider i18nInstance={streami18n}>
+        <Chat client={chatClient} i18nInstance={streami18n}>
+          {children}
+        </Chat>
+      </OverlayProvider>
+    </GestureHandlerRootView>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+});
+```
+
+:::note
+It is essential for you to wrap the [`GestureHandlerRootView`](https://getstream.io/chat/docs/sdk/reactnative/basics/troubleshooting/#touchables-not-working), [`OverlayProvider`](https://getstream.io/chat/docs/sdk/reactnative/core-components/overlay-provider/) and [`Chat`](https://getstream.io/chat/docs/sdk/reactnative/core-components/chat/) component on the root of your chat integration hierarchy.
+:::
```

Comment:
We should be clear about `GestureHandlerRootView' being at the root (index.js)

and others in the screen that has chat 

this text seems to say that everything must be  in one place

Reply:
this is clear in https://getstream.io/chat/react-native-chat/tutorial/#setup-basic-navigation-stack

Reply:
It's not essential that you need to have it on the root unless you use anything around gesture in some other hierarchy other than where the chat is being used.

Reply:
Actually no. Gesture handler must be on the root level. essential!